nullenum_test.go

```diff
@@ -0,0 +1,951 @@
+	}{
+		{
+			name:  "Valid Enum",
+			input: NullEnum[float64]{Value: 42.1, Valid: true},
+			want:  []byte(`42.1`),
+		},
+		{
+			name:  "Invalid Enum",
+			input: NullEnum[float64]{Valid: false},
+			want:  []byte("null"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			got, err := tt.input.MarshalJSON()
+			if err != nil {
+				t.Errorf("MarshalJSON() error = %v", err)
+			}
+			if diff := cmp.Diff(tt.want, got); diff != "" {
+				t.Errorf("MarshalJSON() mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+// TestUnmarshalJSON validates UnmarshalJSON.
+func TestNullEnum_UnmarshalJSON_string(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		input   []byte
+		want    NullEnum[string]
+		wantErr bool
+	}{
+		{
+			name:  "Valid Enum",
+			input: []byte(`"enumValue"`),
+			want:  NullEnum[string]{Value: "enumValue", Valid: true},
+		},
+		{
+			name:  "Null JSON",
+			input: []byte(`null`),
+			want:  NullEnum[string]{Value: "", Valid: false},
+		},
+		{
+			name:    "Invalid JSON",
+			input:   []byte(`"enumValue`), // missing closing quote
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			var n NullEnum[string]
+			err := n.UnmarshalJSON(tt.input)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("UnmarshalJSON() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if !tt.wantErr {
+				if diff := cmp.Diff(tt.want, n); diff != "" {
+					t.Errorf("UnmarshalJSON() mismatch (-want +got):\n%s", diff)
+				}
+			}
+		})
+	}
+}
+
+// TestUnmarshalJSON validates UnmarshalJSON.
+func TestNullEnum_UnmarshalJSON_int(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		input   []byte
+		want    NullEnum[int]
+		wantErr bool
+	}{
+		{
+			name:  "Valid Enum",
+			input: []byte(`42`),
+			want:  NullEnum[int]{Value: 42, Valid: true},
+		},
+		{
+			name:  "Null JSON",
+			input: []byte(`null`),
+			want:  NullEnum[int]{Value: 0, Valid: false},
+		},
+		{
+			name:    "Invalid JSON",
+			input:   []byte(`"enumValue`), // missing closing quote
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			var n NullEnum[int]
+			err := n.UnmarshalJSON(tt.input)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("UnmarshalJSON() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if !tt.wantErr {
+				if diff := cmp.Diff(tt.want, n); diff != "" {
+					t.Errorf("UnmarshalJSON() mismatch (-want +got):\n%s", diff)
+				}
+			}
+		})
+	}
+}
+
+// TestUnmarshalJSON validates UnmarshalJSON.
+func TestNullEnum_UnmarshalJSON_int64(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		input   []byte
+		want    NullEnum[int64]
+		wantErr bool
+	}{
+		{
+			name:  "Valid Enum",
+			input: []byte(`42`),
+			want:  NullEnum[int64]{Value: 42, Valid: true},
+		},
+		{
+			name:  "Null JSON",
+			input: []byte(`null`),
```

Comment:
```suggestion
			input: []byte("null")},
```

Recommend this for consistency with the corresponding MarshalJSON test case.